packages/regl-worldview/src/commands/Lines.js

```diff
@@ -375,52 +486,52 @@ const lines = (regl: any) => {
     positionBuffer2({ data: pointArray, usage: "dynamic" });
 
     const monochrome = !(props.colors && props.colors.length);
-    let colors;
-    if (monochrome) {
-      if (shouldConvert(props.color)) {
-        colors = monochromeColors.fill(props.color ? toRGBA(props.color) : DEFAULT_MONOCHROME_COLOR);
-      } else {
-        colors = monochromeColors.fill(props.color || DEFAULT_MONOCHROME_COLOR);
-      }
-    } else {
-      if (shouldConvert(props.colors)) {
-        colors = props.colors.map(toRGBA);
-      } else {
-        colors = props.colors.slice();
-      }
-      if (shouldClose) {
-        colors.push(colors[0]);
-      }
-    }
-    colorBuffer({ data: colors, usage: "dynamic" });
+    fillColorArray(props.color, props.colors, monochrome, shouldClose);
+    colorBuffer({ data: colorArray, usage: "dynamic" });
 
     const joined = primitive === "line strip";
     const effectiveNumPoints = numInputPoints + (shouldClose ? 1 : 0);
     const instances = joined ? effectiveNumPoints - 1 : Math.floor(effectiveNumPoints / 2);
 
+    // fill instanced pose buffers
+    const { poses } = props;
+    const hasInstancedPoses = !!poses && poses.length > 0;
+    const error = fillPoseArrays(instances, poses, hasInstancedPoses);
+    if (error) {
+      console.error(error);
+      return;
+    }
+    posePositionBuffer({ data: positionArray, usage: "dynamic" });
+    poseRotationBuffer({ data: rotationArray, usage: "dynamic" });
+
     render(debug, () => {
-      command({
-        ...props,
-        joined,
-        primitive: "triangle strip",
-        alpha: debug ? 0.2 : 1,
-        monochrome,
-        instances,
-        scaleInvariant,
-      });
-      if (debug) {
-        command({
-          ...props,
+      // Use Object.assign because it's actually faster than babel's object spread polyfill.
```

Comment:
is it possible to use neither? Just construct a single object w/ all known keys, taking any existing key from `props` and providing the default in the `{ }` part of the `Object.assign` call?  I reckon that's far faster than either more dynamic route.

Reply:
It is possible to use neither but I tested object.assign and it seems to take essentially no time so I feel pretty comfortable about using it - the tradeoff of requiring developers to enumerate keys here kinda sucks